streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamTaskTest.java

```diff
@@ -1144,4 +1146,31 @@ protected void flushState() {
         );
     }
 
+    @Test(expected = IllegalStateException.class)
+    public void shouldThrowOnCleanCloseTaskWhenEosEnabledIfTransactionInFlight() {
+        task = createStatelessTask(createConfig(true));
+        task.close(true, false);
```

Comment:
if if should throw here, the lines below can be removed

Also, we should not use `expected` annotation because if `createStatelessTask()` or `createConfig()` throw, the test should fail but not pass.

rewrite to:
```
try {
    task.close(true, false);
    fail("should have throw IllegalStateException");
} catch (final IllegalStateException expected) {
    // pass
}
```

Reply:
Fixed